polars/polars-io/src/lib.rs

```diff
@@ -134,10 +135,29 @@ pub(crate) fn finish_reader<R: ArrowReader>(
             }
         }
     }
-    let mut df = accumulate_dataframes_vertical(parsed_dfs)?;
 
-    // Aggregations must be applied a final time to aggregate the partitions
-    apply_aggregations(&mut df, aggregate)?;
+    let df = {
+        if parsed_dfs.is_empty() {
+            // Create an empty dataframe with the correct data types
+            let empty_cols = arrow_schema
+                .fields
+                .iter()
+                .map(|fld| {
+                    Series::try_from((
+                        fld.name.as_str(),
+                        Arc::from(new_empty_array(fld.data_type.clone())),
+                    ))
+                })
+                .collect::<Result<_>>()?;
+            DataFrame::new(empty_cols)?
+        } else {
+            // If there any rows, accumulate them into a df
+            let mut df = accumulate_dataframes_vertical(parsed_dfs)?;
+            // Aggregations must be applied a final time to aggregate the partitions
+            apply_aggregations(&mut df, aggregate)?;
```

Comment:
@ritchie46 Just to double-check if this part is correct.

Reply:
Looks good! :+1: